gammapy/maps/core.py

```diff
@@ -843,7 +843,7 @@ def get_image_by_idx(self, idx):
             raise ValueError("Tuple length must equal number of non-spatial dimensions")
 
         # Only support scalar indices per axis
-        idx = tuple([int(_) for _ in idx])
+        idx = tuple([int(np.squeeze(_)) for _ in idx])
```

Comment:
I have used `squeezed` to let the function fail if the quantity is not reducible to a single value rather than let it continue silently. I don't know what is the best behavior

Reply:
As I said the, I think the recommended syntax by Numpy is `_.item()`. This will fail if the array contains more than one element, otherwise it return the the Python scalar. So the additional call to `int()` can be skipped.